### ISSUE TITLE
disableAnimations not disabling all animations

### ISSUE THREAD

**Issue by jack775544** [microsoft/terminal#8405]
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.19042.630]
Windows Terminal version (if applicable): 1.5.3242.0, 1.4.3141.0

Any other software?
No
```

# Steps to reproduce

Adding `"disableAnimations": true,` to my profile does not disable command palette animations or the animation that plays when opening the dropdown menu to open a new terminal. 

I have found that globally disabling animations in windows will disable these animations.

# Expected behavior

Those animations stop playing.

# Actual behavior

The animations continue playing.

**Comment by DHowett** [comment]
Just to confirm: you are putting `disableAnimations` into the _profile_? If so, it will not work there. It must be at the top level of the settings file outside of any profiles.

We released 1.5.3282 with a fix for `disableAnimations`, as well.

Only version 1.5+ even has animations, so I would be concerned if you are seeing them on 1.4.

**Comment by jack775544** [comment]
Yep, placing it on the top level of the settings, my apologies on the wording. I am seeing pane animations being correctly disabled when opening a new pane however the command palette still retains them.

![wtanimations](https://user-images.githubusercontent.com/8287765/100306592-803b6f00-2fef-11eb-96a4-dac5cdc17a67.gif)

In that above gif you can see when I am typing and deleting words the animation plays to remove and add to the list. Disabling windows animations globally will prevent it from occurring however.

**Comment by DHowett** [comment]
Oh! Wow, I forgot that there were animations in that component. It somewhat makes sense; we aren’t driving those and they are provided by the system. We should still be able to turn them off.
Good catch!

**Comment by zadjii-msft** [comment]
I'm not sure that an app is actually able to disable implicit animations on itself... Maybe @mdtauk might know?

**Comment by mdtauk** [comment]
As far as I know, the intention of these controls, are to respond to the system level setting to disable animation.

I'm not aware of a WinUI level API which can manage this.  And to achieve it may require altering control templates to remove transitions and storyboards which drive these.

You should probably contact the WinUI team, and ask them what the best approach would be.

**Comment by zadjii-msft** [comment]
That's what I was afraid of.  The setting certainly implies that it'll disable all animations within the app, but we can't control all of them, only the ones for the controls we author ourselves. 

Something like `disablePaneAnimations` would I guess make more sense. Let's entertain a hypothetical though, say we added an animation to the opening of the command palette. Then we'd need something like `disableCommandPaletteAnimations`, but that wouldn't disable all animations in the palette, just the opening one. Hmmmmmm

**Comment by zakius** [comment]
>You should probably contact the WinUI team, and ask them what the best approach would be.

best approach would be not having animations enabled by default system-wide, but seeing the course they've taken that's unfortunately out of question, second best allowing components to be programmatically configured if they should use system-wide setting or force disable (no way to force enable to prevent annoying apps enforcing animations)

**Comment by dmex** [comment]
@zadjii-msft 

> we can't control all of them, only the ones for the controls we author ourselves.

You can override the ResourceDictionary property of those controls before adding them to XamlRoot. Specify the same ResourceKey but with values that disable the animations. This is how you were expected to disable animations and style the 'native' builtin controls by adding your own keys into App.Resources and for other cases with OverrideMetadata.

You can also do this globally for the application by QueryInterface for the global ResourceDictionary which exists in XamlIslands but it's not well documented.

**Comment by tommdq** [comment]
I'm having the same problem when using "disableAnimations" in the settings.json file.

**Comment by carlos-zamora** [comment]
PowerToys recently merged in a possibly relevant PR: https://github.com/microsoft/PowerToys/pull/28128

We may be able to do what they did.
